shell/common/rasterizer.cc

```diff
@@ -321,37 +321,43 @@ static sk_sp<SkSurface> CreateSnapshotSurface(GrContext* surface_context,
   return SkSurface::MakeRaster(image_info);
 }
 
-static sk_sp<SkData> ScreenshotLayerTreeAsImage(
+sk_sp<SkData> Rasterizer::ScreenshotLayerTreeAsImage(
     flutter::LayerTree* tree,
     flutter::CompositorContext& compositor_context,
     GrContext* surface_context,
     bool compressed) {
-  // Attempt to create a snapshot surface depending on whether we have access to
-  // a valid GPU rendering context.
-  auto snapshot_surface =
-      CreateSnapshotSurface(surface_context, tree->frame_size());
-  if (snapshot_surface == nullptr) {
-    FML_LOG(ERROR) << "Screenshot: unable to create snapshot surface";
-    return nullptr;
-  }
-
-  // Draw the current layer tree into the snapshot surface.
-  auto* canvas = snapshot_surface->getCanvas();
-
-  // There is no root surface transformation for the screenshot layer. Reset the
-  // matrix to identity.
-  SkMatrix root_surface_transformation;
-  root_surface_transformation.reset();
+  // Prepare an image from the surface, this image may potentially be on the
```

Comment:
Despite the big diff, here I only added 
```cpp
const auto screen_shot_provider = surface_->GetScreenShotProvider();
  if (snapshot_surface == nullptr) {	  if (screen_shot_provider != nullptr) {
    FML_LOG(ERROR) << "Screenshot: unable to create snapshot surface";	    potentially_gpu_snapshot = screen_shot_provider->TakeScreenShot();
    return nullptr;	  
}
```
and put everything into an "else" block.